.github/workflows/docker-scan.yml

```diff
@@ -0,0 +1,55 @@
+name: Build and Scan Docker Images
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-and-scan:
+    name: ${{ matrix.folder }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        folder: ["curl", "httpd", "locust", "nginx", "wireshark"]
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4.2.2
+
+      - name: Log in to Docker Hub
+      #  Required for Docker Scout
+        uses: docker/login-action@v3.3.0
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build Docker Image
+        run: |
+          FILES=$(find ./${{ matrix.folder }} -type f -iname 'dockerfile*')
+          if [ -z "$FILES" ]; then
+            echo "No Dockerfiles found in folder: ${{ matrix.folder }}. Skipping build."
+            exit 1
+          fi
+
+          for FILE in $FILES; do
+            IMAGE_NAME="${{ matrix.folder }}-$(basename $FILE | tr '[:upper:]' '[:lower:]' | tr -cd '[:alnum:]-')"
+            echo "Building Docker image: $IMAGE_NAME using $FILE"
+            docker build -t $IMAGE_NAME -f $FILE ./${{ matrix.folder }}
```

Comment:
This logic has a flaw @Hawazyn : If finding a Dockerfile in a subdirectory (such as nginx/fulltest), the subsequent `docker build`command does not switch CWD to that subdirectory, leading to files not being found. IMO this is the reason for the persistent CI failure, e.g., [here](https://github.com/open-quantum-safe/oqs-demos/actions/runs/12397922498/job/34649139160). Please let me know whether you'd provide a PR to fix for this or whether I should do to get CI status back to green.

Reply:
Hey @baentsch, I’ll look into this issue today and provide a fix PR.

Reply:
On closer look, I'm wondering whether this can be improved in general: Am I right thinking that this job duplicates the tasks in the other CI workflows (building images)? Wouldn't it be better in general to only add a scan step to each separate image build as and when done anyway? This would be much less resource intensive. OK for you to change things this way? This would also do away with this separate file entirely (and thus automatically solve the problem above): OK for you @Hawazyn ?

Reply:
I agree with combining the scan with image builds for efficiency. Should I proceed to apply the changes and remove the docker-scan file?

Reply:
If you have time for that, that would be welcome.

Reply:
I’ll work on this, but I kindly ask for some additional time as I am currently managing other priorities, in addition to work related to Docker files and their documentation. I truly appreciate your patience and understanding, and I will keep you updated as I make progress.

Reply:
No urgency -- would you mind if I did it to save you the hassle and let you focus on what you're already doing?

Reply:
Thank you for offering to help! I can see you already have a great deal to manage at the moment, and while I am perfectly happy to continue with this, I do not wish to add to your workload unnecessarily. My tasks are quite manageable, so please let me know how you would prefer to proceed. I am happy to adjust to whatever suits you best.

Reply:
I sincerely apologize for being unable to address the workflow scanning integration as planned due to unforeseen circumstances. Unfortunately, I will be unavailable for the next month and unable to take on new tasks during this time. However, I am committed to finalizing this PR to ensure the current work is completed before my absence.